connect/cli/plugins/project/report/validations.py

```diff
@@ -0,0 +1,161 @@
+
+#  Copyright © 2022 CloudBlue. All rights reserved.
+
+import json
+import os
+import sys
+import inspect
+import importlib
+
+from connect.reports.parser import parse
+from connect.reports.validator import (
+    validate,
+    validate_with_schema,
+)
+from connect.cli.plugins.project.validators import (
+    get_code_context,
+    load_project_toml_file,
+    ValidationItem,
+    ValidationResult,
+)
+from connect.cli.core.utils import get_last_cli_version
+
+from poetry.core.semver import parse_constraint, Version
+
+
+def validate_pyproject_toml(project_dir, context):
+    messages = []
+    data = load_project_toml_file(project_dir)
+    if isinstance(data, ValidationResult):
+        return data
+
+    descriptor_file = os.path.join(project_dir, 'pyproject.toml')
+    dependencies = data.get('tool', {}).get('poetry', {}).get('dev-dependencies', {})
+
+    if not dependencies or 'connect-cli' not in dependencies:
+        messages.append(
+            ValidationItem(
+                'WARNING',
+                'No development dependency on *connect-cli* has been found.',
+                descriptor_file,
+            ),
+        )
+    if 'connect-cli' in dependencies:
+        project_version = parse_constraint(dependencies['connect-cli'])
+        last_version = get_last_cli_version()
+        last_version_obj = Version(*last_version.split('.'))
+        if not project_version.allows(last_version_obj):
+            messages.append(
+                ValidationItem(
+                    'WARNING',
+                    f'The development dependency *connect-cli* is not the latest version which is {last_version}.',
```

Comment:
I suggest the following message:
The version range {dependencies['connect-cli']} specified in your pyproject.toml does not include the lastest connect-cli version which is {last_version}.